Compilation failure in wxMain.cpp (line 225)

UCBLogo 6.2.3, Linux, wxGTK 3.0.5

I get the following error when trying to compile:

wxMain.cpp: In function ‘void getExecutableDir(char*, int)’:
wxMain.cpp:225:17: error: cannot convert ‘wxString’ to ‘const char*’
  225 |   strncpy(path, executable_dir, maxlen);
      |                 ^~~~~~~~~~~~~~
      |                 |
      |                 wxString


I know very little about wxWidgets programming, but after looking at some of the documentation for wxString, I changed line 225 to the following:

strncpy(path, (const char*)executable_dir.mb_str(wxConvUTF8), maxlen);

UCBLogo now compiles and seems to work correctly. I don't want to claim this is necessarily the correct or best way to solve this problem, but clearly some change is needed.

Update: I've also tried compiling against wxGTK 3.2.2.1 and can confirm the same problem happens (though from reading the wxWidgets documentation on wxString, I didn't think this would make a difference).

--- a/wxMain.cpp
+++ b/wxMain.cpp
@@ -222,7 +222,7 @@ extern "C" int wxUnget_c(int c, FILE * f) {
 extern "C" void getExecutableDir(char * path, int maxlen) {
   wxString executable = wxStandardPaths::Get().GetExecutablePath();
   wxString executable_dir =  wxFileName(executable).GetPath();
-  strncpy(path, executable_dir, maxlen);
+  strncpy(path, (const char*)executable_dir.mb_str(wxConvUTF8), maxlen);
   path[maxlen - 1] = '\0';
 }
 

